dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java

```diff
@@ -0,0 +1,104 @@
+     * @param workflowVersionDAO Workflow Version data access object
+     * @param user user authenticated to update the workflow
+     * @param workflowVersionId the id of the workflow version
+     */
+    public static WorkflowVersion getAndCheckWorkflowVersionResource(WorkflowResource workflowResource, WorkflowDAO workflowDAO,
+            WorkflowVersionDAO workflowVersionDAO, User user, Long workflowVersionId) {
+        final WorkflowVersion workflowVersion = workflowVersionDAO.findById(workflowVersionId);
+        if (workflowVersion == null) {
+            LOG.error("Could not find workflow version using the workflow version id: " + workflowVersionId);
+            throw new CustomWebApplicationException("Workflow version not found when searching with id: " + workflowVersionId, HttpStatus.SC_BAD_REQUEST);
+        }
+
+        Long workflowId = getWorkflowId(workflowDAO, workflowVersionId);
+        Workflow workflow = workflowDAO.findById(workflowId);
+        workflowResource.checkEntry(workflow);
+        workflowResource.checkUserCanUpdate(user, workflow);
+        return workflowVersion;
+    }
+
+    /**
+     * Add aliases to a Workflow Version
+     * and check that they are valid before adding them:
+     * 1. If admin/curator, then no limit on prefix
+     * 2. If blockFormat false, then no limit on format
+     * @param workflowResource Workflow resource object
+     * @param workflowDAO Workflow data access object
+     * @param workflowVersionDAO Workflow Version data access object
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param id the id of the Entry
+     * @param aliases a comma separated string of aliases
+     * @param blockFormat if true don't allow specific formats
+     * @return the Workflow Version
+     */
+    public static WorkflowVersion addWorkflowVersionAliasesAndCheck(WorkflowResource workflowResource, WorkflowDAO workflowDAO,
+            WorkflowVersionDAO workflowVersionDAO, User user, Long id, String aliases, boolean blockFormat) {
+        WorkflowVersion workflowVersion = getAndCheckWorkflowVersionResource(workflowResource, workflowDAO, workflowVersionDAO, user, id);
+        Set<String> oldAliases = workflowVersion.getAliases().keySet();
+        Set<String> newAliases = Sets.newHashSet(Arrays.stream(aliases.split(",")).map(String::trim).toArray(String[]::new));
+
+        AliasableResourceInterface.checkAliases(newAliases, user, blockFormat);
+
+        Set<String> duplicateAliasesToAdd = Sets.intersection(newAliases, oldAliases);
+        if (!duplicateAliasesToAdd.isEmpty()) {
+            String dupAliasesString = String.join(", ", duplicateAliasesToAdd);
+            throw new CustomWebApplicationException("Aliases " + dupAliasesString + " already exist; please use unique aliases",
+                    HttpStatus.SC_BAD_REQUEST);
+        }
+
+        newAliases.forEach(alias -> workflowVersion.getAliases().put(alias, new Alias()));
```

Comment:
This was like this before, but I just noticed now... why is there an empty alias being added? Alias has a content field, should that not be filled in?

Reply:
I don't know; I don't think it is filled in anywhere else...

Reply:
What ends up in the table?

Reply:
The content field is empty

Reply:
If it's empty, what's the point of having the field? Does this behavior happen for other types of aliases as well?

I feel like this needs to be understood before merging. We shouldn't be creating tables or columns that aren't used.

Reply:
It was always there...

Reply:
Comment in Alias.java is 
/**
 * Currently stores alias information for Collections, Organizations, Tools and Workflows.
 * For now its just blank, but can be expanded with additional information on the aliases (such as if they
 * point at dockstore mirrors)
 */

Reply:
public Entry aliases(Map<String, Alias> aliases) {
    this.aliases = aliases;
    return this;
  }

  public Entry putAliasesItem(String key, Alias aliasesItem) {
    if (this.aliases == null) {
      this.aliases = new HashMap<>();
    }
    this.aliases.put(key, aliasesItem);
    return this;
  }

Reply:
webservice_test=# \d collection_alias
                       Table "public.collection_alias"
    Column    |            Type             | Collation | Nullable | Default 
--------------+-----------------------------+-----------+----------+---------
 id           | bigint                      |           | not null | 
 content      | text                        |           |          | 
 dbcreatedate | timestamp without time zone |           |          | 
 dbupdatedate | timestamp without time zone |           |          | 
 alias        | text                        |           | not null | 
Indexes:
    "collection_alias_pkey" PRIMARY KEY, btree (id, alias)
    "unique_col_aliases" UNIQUE CONSTRAINT, btree (alias)
Foreign-key constraints:
    "fk_collection_alias" FOREIGN KEY (id) REFERENCES collection(id)

webservice_test=# select * from collection_alias;
 id | content | dbcreatedate | dbupdatedate | alias 
----+---------+--------------+--------------+-------
(0 rows)

Reply:
webservice_test=# select * from collection_alias;
 id | content | dbcreatedate | dbupdatedate | alias 
----+---------+--------------+--------------+-------
(0 rows)

webservice_test=# select * from workflowversion_alias;
  id   | content | dbcreatedate | dbupdatedate |         alias         
-------+---------+--------------+--------------+-----------------------
 19356 |         |              |              | 10.5072-zenodo.438221
 18272 |         |              |              | 10.5072-zenodo.438222
(2 rows)